### DISCUSSION TITLE
Nesting server and client components

### DISCUSSION THREAD

**Discussion by AdiGutner** [vercel/next.js#50597] [Help]
I want to create a tabs component where every tab has an id, and when a tab is clicked I want to make an API call using the tab id to display the tab's content. I'm struggling with how to keep the state of the currently open tab in the client tabs component, as well as how to pass it to the content server component.

**Comment by devjiwonchoi** [comment]
As far as I know, if you try to `fetch` an API, the end of `state`'s life cycle is inevitable since `fetch` is an asynchronous call.

You can try to use [`use`](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises.md#usepromise) since the state will be "terminated" and "re-executed" from the previous `state` 
as if the `state` is paused and resumed.

> the runtime for use maintains an internal state machine to suspend and resume; but from the perspective of the component author, it looks and feels like a sequential function:

Certainly `use` is also an asynchronous call, but it works inside a non-async function.
> use is designed to provide much the same programming model as async/await, while still working inside regular (non-async) function components and Hooks.


Example:
```
function Note({id}) {
  // This fetches a note asynchronously, but to the component author it looks
  // like a synchronous operation.
  const note = use(fetchNote(id));
  return (
    <div>
      <h1>{note.title}</h1>
      <section>{note.body}</section>
    </div>
  );
}
```

For more information about `use`, look [here](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises.md#usepromise).

Even though the `use` in Client Component is still unstable,
> While you can try it out, it is not yet stable. We'll keep these docs updated to reflect the latest developments.

Next.js is officially recommending to use `use` in Client Components.
> This new data fetching model is currently being developed by the React team. We recommend reading the [support for promises React RFC](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises.md) which introduces async and await in Server Components and a new use() hook for Client Components.

ref: https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#use-in-client-components
